cluster/provision_master.sh

```diff
@@ -0,0 +1,130 @@
+    retry master 'netstat -anp | grep LISTEN | grep -w 8091' # query-engine
+    retry master 'netstat -anp | grep LISTEN | grep -w 6379' # redis
+    retry master 'netstat -anp | grep LISTEN | grep -w 8143' # WebUI
+    retry master 'netstat -anp | grep LISTEN | grep -w 8070' # WebUI
+    retry master 'netstat -anp | grep LISTEN | grep -w 3000' # WebUI
+#   retry master 'netstat -anp | grep LISTEN | grep -w 5998' # discovery
+#   retry master 'netstat -anp | grep LISTEN | grep -w 8094' # dns
+#   retry master 'netstat -anp | grep LISTEN | grep -w 53' # named
+}
+
+# Provision controller
+function provision_controller() {
+    cmd='docker ps | grep contrail-api | grep -v pause | awk "{print \"docker exec \" \$1 \" curl -s https://raw.githubusercontent.com/Juniper/contrail-controller/R2.20/src/config/utils/provision_control.py -o /tmp/provision_control.py\"}" | sudo sh'
+    master $cmd
+    cmd='docker ps | grep contrail-api | grep -v pause | awk "{print \"docker exec \" \$1 \" curl -s https://raw.githubusercontent.com/Juniper/contrail-controller/R2.20/src/config/utils/provision_bgp.py -o /tmp/provision_bgp.py\"}" | sudo sh'
+    master $cmd
+    cmd='docker ps | grep contrail-api | grep -v pause | awk "{print \"docker exec \" \$1 \" python /tmp/provision_control.py  --router_asn 64512 --host_name `hostname` --host_ip `hostname --ip-address` --oper add --api_server_ip `hostname --ip-address` --api_server_port 8082\"}" | sudo sh'
+    master $cmd
+}
+
+# Provision link local service
+function provision_linklocal() {
+    cmd='docker ps | grep contrail-api | grep -v pause | awk "{print \"docker exec \" \$1 \" curl -s https://raw.githubusercontent.com/Juniper/contrail-controller/R2.20/src/config/utils/provision_linklocal.py -o /tmp/provision_linklocal.py\"}" | sudo sh'
+    master $cmd
+    cmd='docker ps | grep contrail-api | grep -v pause | awk "{print \"docker exec \" \$1 \" python /tmp/provision_linklocal.py --api_server_ip `hostname --ip-address` --api_server_port 8082 --linklocal_service_name kubernetes --linklocal_service_ip 10.0.0.1 --linklocal_service_port 8080 --ipfabric_service_ip `hostname --ip-` --ipfabric_service_port 8080 --oper add\"}" | sudo sh'
+    master $cmd
+}
+
+# Setup kube dns endpoints
+function setup_kube_dns_endpoints() {
+    master kubectl --namespace=kube-system create -f /etc/kubernetes/addons/kube-ui/kube-ui-endpoint.yaml || true
+    master kubectl --namespace=kube-system create -f /etc/kubernetes/addons/kube-ui/kube-ui-svc-address.yaml || true
+}
+
+# Setup contrail manifest files under kubernetes
+function setup_contrail_manifest_files() {
+    cmd='wget -qO - https://raw.githubusercontent.com/rombie/contrail-kubernetes/manifests/cluster/manifests.hash | grep -v kube-network-manager | grep -v contrail-vrouter-agent | awk "{print \"https://raw.githubusercontent.com/rombie/contrail-kubernetes/manifests/cluster/\"\$1}" | xargs -n1 sudo wget -q --directory-prefix=/etc/contrail/manifests --continue'
```

Comment:
Salt can copy, but most of the times docker hangs when it tries to download
all at once. kubernetes also have seen that issue and hence do this. I have
found success in bringing all correctly, only first doing "docker pull",
and then copy the manifest files to /etc/kubernetes/manifests.

https://github.com/Juniper/kubernetes/blob/master/cluster/saltbase/salt/kube-master-addons/kube-master-addons.sh

This also reduces the diffs quite a bit, with just 3 manifest files for
contrail (one for kube-network-manager and 2 for the master and minion
contrail-provision scripts)


On Wed, Sep 2, 2015 at 4:28 PM, Pedro Marques <notifications@github.com>
wrote:

> In cluster/provision_master.sh
> <https://github.com/Juniper/contrail-kubernetes/pull/40#discussion_r38598070>
> :
>
> > +function provision_linklocal() {
> > +    cmd='docker ps | grep contrail-api | grep -v pause | awk "{print \"docker exec \" \$1 \" curl -s https://raw.githubusercontent.com/Juniper/contrail-controller/R2.20/src/config/utils/provision_linklocal.py -o /tmp/provision_linklocal.py\"}" | sudo sh'
> > +    master $cmd
> > +    cmd='docker ps | grep contrail-api | grep -v pause | awk "{print \"docker exec \" \$1 \" python /tmp/provision_linklocal.py --api_server_ip `hostname --ip-address` --api_server_port 8082 --linklocal_service_name kubernetes --linklocal_service_ip 10.0.0.1 --linklocal_service_port 8080 --ipfabric_service_ip `hostname --ip-` --ipfabric_service_port 8080 --oper add\"}" | sudo sh'
> > +    master $cmd
> > +}
> > +
> > +# Setup kube dns endpoints
> > +function setup_kube_dns_endpoints() {
> > +    master kubectl --namespace=kube-system create -f /etc/kubernetes/addons/kube-ui/kube-ui-endpoint.yaml || true
> > +    master kubectl --namespace=kube-system create -f /etc/kubernetes/addons/kube-ui/kube-ui-svc-address.yaml || true
> > +}
> > +
> > +# Setup contrail manifest files under kubernetes
> > +function setup_contrail_manifest_files() {
> > +    cmd='wget -qO - https://raw.githubusercontent.com/rombie/contrail-kubernetes/manifests/cluster/manifests.hash | grep -v kube-network-manager | grep -v contrail-vrouter-agent | awk "{print \"https://raw.githubusercontent.com/rombie/contrail-kubernetes/manifests/cluster/\"\$1}" | xargs -n1 sudo wget -q --directory-prefix=/etc/contrail/manifests --continue'
>
> Can't salt copy the manifests directly to the right place ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Juniper/contrail-kubernetes/pull/40/files#r38598070>.
>